### PR TITLE
chore: Create module, Makefile

### PR DESCRIPTION
**Pull request by bhavanki**

### CODE REVIEW

**Comment by bhavanki** [comment]
Supersedes https://github.com/segmentio/go-athena/pull/29 since it's been a long time. Thank you @fzerorubigd for your previous PR!

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module github.com/segmentio/go-athena
+
+go 1.22
```
Comment (alecjacobs5401) [inline go.mod:3]: do we want to heavily restrict the go versions that can consume this package by setting this so high? By setting this, we say that minimum packages must have go 1.22 to consume this, right?

Comment (bhavanki) [reply go.mod:3]: I'm generally OK with it. Go 1.21 and older aren't supported any more, so I'm reluctant to go as far back as 1.19: https://endoflife.date/go We could add a tag for the current HEAD, before this commit, for the benefit of current users when they can't move up to a supported Go version.

I also expect to bump this up to aws-sdk-go-v2 in the near future. That's on 1.21 right now, so I'd be OK going back that far.

Comment (alecjacobs5401) [reply go.mod:3]: Works for me!

**Review by alecjacobs5401 — APPROVED** [review]

**Pull request merged**